### DISCUSSION TITLE
CMake and JSON

### DISCUSSION THREAD

**Discussion by Spahatticat** [nlohmann/json#3870] [Q&A]
I just need some help putting json in my cmake project

**Comment by nlohmann** [comment]
https://json.nlohmann.me/integration/cmake/

**Comment by tomokit0811** [comment]
I failed with "External" described in https://json.nlohmann.me/integration/cmake/:
```
cmake_minimum_required(VERSION 3.1)
project(ExampleProject LANGUAGES CXX)

find_package(nlohmann_json 3.11.2 REQUIRED)

add_executable(example example.cpp)
target_link_libraries(example PRIVATE nlohmann_json::nlohmann_json)
```
Because the nlohman-json is the header-only library, it may require another manner.
I succussed with the following way:
```
find_package(nlohmann_json REQUIRED)
if(nlohmann_json_FOUND)
  include_directories(${nlohmann_json_INCLUDE_DIRS})
else()
  message(STATUS "nlohmann_json NOT Found !")
endif(nlohmann_json_FOUND)
```
I'm glad if this would help TinyRobby.
